### PR TITLE
Update to k8s.io/api/autoscaling/v2

### LINKED ISSUE #104: [Feature] Handle autoscaling/v2 API version for `HorizontalPodAutoscaler`
**Feature (What you would like to be added):**
Add handling for different API versions of `HorizontalPodAutoscaler`

**Motivation (Why is this needed?):**
The autoscaling/v2beta1 API version of HorizontalPodAutoscaler will no longer be served in v1.25.
We have to migrate manifests and API clients to use the autoscaling/v2 API version, available since v1.23.
Ref: [kubernetes announcement](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#horizontalpodautoscaler-v125)

**Approach/Hint to the implement solution (optional):**

### PR DESCRIPTION
**Pull request by voelzmo**
**What this PR does / why we need it**:
Introduce a new HVPA API version which uses `k8s.io/api/autoscaling/v2` instead of `v2beta1`, which is removed in k8s `1.25`. 
This *does not* yet change the HVPA controller logic to no longer access any `autoscalingv2beta1` HPA, so the controller still doesn't run with k8s >=1.25.

**Which issue(s) this PR fixes**:
part of #104 

**Release note**:

```feature user
Introduce a new HVPA API `v1alpha2` which supports the `k8s.io/api/autoscaling/v2` API for HPA configuration 
```

### CODE REVIEW

**Comment by rfranzke** [comment]
Generally, you can use the `envtest` package of `controller-runtime` to run your components, register CRDs and test their logic. Whether it makes sense or not I cannot judge, but if you still have plans to get rid of HVPA entirely then probably not.

**Comment by voelzmo** [comment]
> Generally, you can use the `envtest` package of `controller-runtime` to run your components, register CRDs and test their logic. Whether it makes sense or not I cannot judge, but if you still have plans to get rid of HVPA entirely then probably not.

Ah, sorry, let me rephrase: there are already `envtests` in this project, so adding a new one would be possible. I was wondering if you or others in the team had enough experience with envtesting conversion webhooks to judge if this is significant effort on top of existing tests. I searched a bit and [found indication](https://github.com/kubernetes-sigs/controller-runtime/issues/1882) that the kubebuilder/kustomize setup makes things a bit more difficult than it should be.

**Comment by rfranzke** [comment]
Ah, nope, sorry, we don't have any conversion webhooks yet anywhere else (at least I'm not aware of).

File: main.go [hunk 1]
```diff
@@ -80,13 +84,16 @@ func main() {
 		os.Exit(1)
 	}
 
-	// TODO: Enable webhooks
-	/*
+	if os.Getenv("ENABLE_WEBHOOKS") != "false" {
```
Comment (andrerun) [inline main.go:87]: Just mentioning personal preference, not an issue:
For environment variable name, personally I'd prefer an application-specific prefix to avoid collisions. E.g HVPA_ENABLE_WEBHOOKS. This becomes more relevant in the context of naming consistency, after the application grows and suddenly needs some very collision-prone variable, like ENABLE_LOGGING.

File: apis/autoscaling/v1alpha1/hvpa_types.go [hunk 2]
```diff
@@ -349,3 +350,6 @@ type HvpaList struct {
 /*func init() {
 	SchemeBuilder.Register(&Hvpa{}, &HvpaList{})
 }*/
+
+// Hub marks this version as the API conversion Hub
+func (*Hvpa) Hub() {}
```
Comment (andrerun) [inline apis/autoscaling/v1alpha1/hvpa_types.go:355]: Just a random thought: In the hypothetical long run, the v1alpha2 may turn out to be the more natural hub, since  it's likely to be closer to a hypothetical future version.

Not something I'd change at this point though. Besides, iiuc, it has no practical significance, since it's just a matter of moving that one line of code, which can be done effortlessly at any point in time.

**Review by andrerun — APPROVED** [review]
Looks good. Fix one minor issue in main (wrong log entry value), then ship it.

**Comment by voelzmo** [comment]
Closing this after a conversation with @vlerenc
